src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -186,6 +186,15 @@ private async Task UpdateJobRecordAsync(CancellationToken cancellationToken)
                 queryParametersList.Add(Tuple.Create(KnownQueryParameterNames.ContinuationToken, progress.ContinuationToken));
             }
 
+            if (_exportJobRecord.ExportType == ExportJobType.Patient)
+            {
+                queryParametersList.Add(Tuple.Create(KnownQueryParameterNames.Type, KnownResourceTypes.Patient));
```

Comment:
There is an assumption in `ProcessProgressChange` that the ContinuationToken is added to the end of the `queryParametersList`. Adding _type at the end will break things. We should either add the _type before the ContinuationToken. Or better use a different data structure to hold the `queryParametersList` before we pass it on to the search service.

Reply:
Good catch. I replaced the assumption with a loop to find the continuation token if it exists. Since there will never be more than 5 items in the list I don't think this will affect performance.

Reply:
I agree that looping through the parameter list will not affect performance. But I am pretty sure there will be more than 5 items in the list in the future :)

Reply:
Also, I am wondering whether it's possible to have a unit or integration test that could have caught this issue?

Reply:
Added a new unit test for the scenario